src/snakemake/dag.py

```
@@ -466,9 +481,16 @@ def access_pattern(f):
                             f"{f.storage_object.is_ondemand_eligible} with access "
                             f"patterns {','.join(map(str, file_access_patterns))}"
                         )
-                        tg.create_task(f.retrieve_from_storage())
+
+                        async def retrieve_and_yield():
+                            await f.retrieve_from_storage()
```

Ruff (0.15.21)

[warning] Function definition does not bind loop variable `f`

(B023)